Sources/PostgresKit/Builders/PostgresCreateTypeBuilder.swift

```diff
@@ -0,0 +1,52 @@
+/// Builds `PostgresCreateType` queries.
+///
+///    conn.create(enum: "meal", cases: "breakfast", "lunch", "dinner")
+///        .run()
+///
+/// See `SQLColumnBuilder` and `SQLQueryBuilder` for more information.
+public final class PostgresCreateTypeBuilder: SQLQueryBuilder {
+    /// `CreateType` query being built.
+    public var createType: PostgresCreateType
+    
+    /// See `SQLQueryBuilder`.
+    public var database: SQLDatabase
+    
+    /// See `SQLQueryBuilder`.
+    public var query: SQLExpression {
+        return self.createType
+    }
+    
+    /// Creates a new `PostgresCreateTypeBuilder`.
+    public init(_ createType: PostgresCreateType, on database: SQLDatabase) {
+        self.createType = createType
+        self.database = database
+    }
+}
+
+// MARK: Connection
+
+extension SQLDatabase {
+    /// Creates a new `PostgresCreateTypeBuilder`.
+    ///
+    ///     conn.create(enum: "meal", cases: "breakfast", "lunch", "dinner")...
+    ///
+    /// - parameters:
+    ///     - name: Name of ENUM type to create.
+    ///     - cases: The cases of the ENUM type.
+    /// - returns: `PostgresCreateTypeBuilder`.
+    public func create(enum name: String, cases: String...) -> PostgresCreateTypeBuilder {
```

Comment:
Until Swift gets support for passing arrays into veridic types, the `cases` parameter should probably be of type `[String]`. Otherwise there isn't really a good way of dynamically generating a list of cases to pass in, or even passing cases in from a higher-level method.

Reply:
And this would apply to all the `create(enum:cases:)` methods that you've added. Now, another option would be to just create overloads, so you can pass in either an array or the veridic list.

Reply:
Good point. I would have resorted to using the initializer for `PostgresCreateType` but there would have been no convenient way to hook into `SQLDatabase` so that would have bitten me pretty quickly. There's a bit of precedent for builders elsewhere in Vapor offering variadic overloads of functions (although not with 100% consistency) so I'll go the overload route.